salt/api.sls

```diff
@@ -10,6 +10,9 @@ salt-api:
   {%- if salt_settings.version is defined %}
     - version: {{ salt_settings.version }}
   {%- endif %}
+  {%- if salt_settings.pin_version %}
+    - hold: True
+  {%- endif %}
```

Comment:
I suggest you don't mix pinning and holding under same pillar option. It will be better to use separate pillar setting for hold.

I.e. pillar

```yaml
salt:
  hold_version: false
  update_holds: false
```
and relevant update for state
```yaml
salt-master:
    {% if salt_settings.install_packages %}
  pkg.installed:
    - name: {{ salt_settings.salt_master }}
       {%- if salt_settings.version is defined %}
    - version: {{ salt_settings.version }}
       {%- endif %}
    - hold: {{ salt_settings.hold_version }}
    - update_holds: {{ salt_settings.false }}
```